portal-ui/src/screens/Console/Users/AddUser.tsx

```diff
@@ -0,0 +1,201 @@
+  createStyles({
+    errorBlock: {
+      color: "red"
+    }
+  });
+
+interface IAddUserProps {
+  classes: any;
+  closeModalAndRefresh: () => void;
+  selectedUser: User | null;
+}
+
+interface IAddUserState {
+  addLoading: boolean;
+  addError: string;
+  name: string;
+  email: string;
+}
+
+class AddUser extends React.Component<IAddUserProps, IAddUserState> {
+  state: IAddUserState = {
+    addLoading: false,
+    addError: "",
+    name: "",
+    email: ""
+  };
+
+  componentDidMount(): void {
+    const { selectedUser } = this.props;
+    if (selectedUser !== null) {
+      this.setState({
+        name: selectedUser.name,
+        email: selectedUser.email
+      });
+    }
+  }
+
+  saveRecord(event: React.FormEvent) {
+    event.preventDefault();
+    const { name, addLoading, email } = this.state;
+    const { selectedUser } = this.props;
+    if (addLoading) {
+      return;
+    }
+    this.setState({ addLoading: true }, () => {
+      if (selectedUser !== null) {
+        api
+          .invoke("PUT", `/api/v1/users/${selectedUser.id}`, {
+            id: selectedUser.id,
+            name: name,
+            email: email
+          })
+          .then(res => {
+            this.setState(
+              {
+                addLoading: false,
+                addError: ""
+              },
+              () => {
+                this.props.closeModalAndRefresh();
+              }
+            );
+          })
+          .catch(err => {
+            this.setState({
+              addLoading: false,
+              addError: err
+            });
+          });
+      } else {
+        api
+          .invoke("POST", "/api/v1/users", {
+            name: name,
+            email: email
+          })
+          .then(res => {
+            this.setState(
+              {
+                addLoading: false,
+                addError: ""
+              },
+              () => {
+                this.props.closeModalAndRefresh();
+              }
+            );
+          })
+          .catch(err => {
+            this.setState({
+              addLoading: false,
+              addError: err
+            });
+          });
+      }
+    });
+  }
+
+  render() {
+    const { classes, selectedUser } = this.props;
+    const { addLoading, addError, name, email } = this.state;
+
+    return (
+      <form
+        noValidate
+        autoComplete="off"
+        onSubmit={(e: React.FormEvent<HTMLFormElement>) => {
+          this.saveRecord(e);
+        }}
+      >
+        <Grid container>
+          <Grid item xs={12}>
+            {selectedUser !== null ? (
+              <Title>Edit User</Title>
+            ) : (
+              <Title>Add User</Title>
+            )}
+          </Grid>
+          {addError !== "" && (
+            <Grid item xs={12}>
+              <Typography
+                component="p"
+                variant="body1"
+                className={classes.errorBlock}
+              >
+                {`${addError}`}
```

Comment:
please change `{`${addError}`}` to {addError}

Reply:
done